### PR TITLE
Reduce usages of `com.google.common.collect` in favor of native Java functionality

### PR DESCRIPTION
**Pull request by basil**
A subset of #5344 that is hopefully non-controversial and safe to land in the short term because it avoids any changes that have a risk of regression.

The underlying motivation is to make the code easier to read by using a consistent paradigm as much as possible (namely, JDK classes rather than a mixture of JDK classes and Guava classes).

### CODE REVIEW

File: core/src/main/java/hudson/ExtensionFinder.java
Comment (basil) [inline core/src/main/java/hudson/ExtensionFinder.java:315]: `indices` was declared within this method and goes out of scope at the end of the method, so nothing else can modify it. We do not need to copy it.

File: core/src/main/java/hudson/ExtensionFinder.java
Comment (basil) [inline core/src/main/java/hudson/ExtensionFinder.java:671]: `r` was declared within this method and goes out of scope at the end of the method, so nothing else can modify it. We do not need to copy it.

File: core/src/main/java/hudson/model/AbstractBuild.java
Comment (basil) [inline core/src/main/java/hudson/model/AbstractBuild.java:708]: `r` was declared within this method and goes out of scope at the end of the method, so nothing else can modify it. We do not need to copy it.

File: core/src/main/java/hudson/model/DependencyGraph.java
Comment (basil) [inline core/src/main/java/hudson/model/DependencyGraph.java:221]: `builder` was declared within this method and goes out of scope at the end of the method, so nothing else can modify it. We do not need to copy it.

File: core/src/main/java/hudson/model/Fingerprint.java
Comment (basil) [inline core/src/main/java/hudson/model/Fingerprint.java:1181]: The local variable `transientFacets` was declared within this method and goes out of scope at the end of the method, so nothing else can modify it. We do not need to copy it.

File: core/src/main/java/hudson/model/queue/MappingWorksheet.java
Comment (basil) [inline core/src/main/java/hudson/model/queue/MappingWorksheet.java:373]: The local variable `executors` was declared within this method and goes out of scope at the end of the method, so nothing else can modify it. We do not need to copy it.

File: core/src/main/java/hudson/model/queue/MappingWorksheet.java
Comment (basil) [inline core/src/main/java/hudson/model/queue/MappingWorksheet.java:390]: The local variable `works` was declared within this method and goes out of scope at the end of the method, so nothing else can modify it. We do not need to copy it.

File: core/src/main/java/hudson/tasks/Fingerprinter.java
Comment (basil) [inline core/src/main/java/hudson/tasks/Fingerprinter.java:464]: The local variable `m` was declared within this method and goes out of scope at the end of the method, so nothing else can modify it. We do not need to copy it.

File: core/src/main/java/jenkins/agents/WebSocketAgents.java
Comment (basil) [inline core/src/main/java/jenkins/agents/WebSocketAgents.java:95]: `properties` was declared within this method and goes out of scope at the end of the method, so nothing else can modify it. We do not need to copy it.

File: core/src/main/java/jenkins/model/Jenkins.java
Comment (basil) [inline core/src/main/java/jenkins/model/Jenkins.java:2577]: `properties` was declared within this method and goes out of scope at the end of the method, so nothing else can modify it. We do not need to copy it.

**Review by timja — APPROVED** [review]

**Review by res0nance — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by StefanSpieker — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
All looks good. It is hard to estimate the potential downsides though. Performance should not be a problem at least, native Java implementations are commonly better on benchmarks for modern Java versions

**Comment by saper** [comment]
Could this be causing https://issues.jenkins.io/browse/JENKINS-65611 that breaks 2.292 for some people?

**Pull request merged**
